crates/router/src/core/errors.rs

```diff
@@ -112,6 +112,32 @@ pub fn http_not_implemented() -> actix_web::HttpResponse<BoxBody> {
     .error_response()
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum KeyManagerClientError {
```

Comment:
KeyManagerError & KeyManagerError enums are not required, since they are created already in common_utils